src/material-experimental/mdc-slider/slider-thumb-decorator.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {Thumb} from '@material/slider';
+
+/**
+ * Handles displaying the slider knobs and their value indicators.
+ */
+@Component({
+  selector: 'mat-slider-start-thumb-decorator, mat-slider-end-thumb-decorator',
+  templateUrl: 'slider-thumb-decorator.html',
+  host: {
+    'class': 'mdc-slider__thumb',
+    '(mouseenter)': '_mouseenter.emit()',
+    '(mouseleave)': '_mouseleave.emit()',
+  },
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MatSliderThumbDecorator {
+  /** Whether the slider is discrete. */
+  @Input() isDiscrete: boolean;
+
+  /** The text content of the value indicator for a discrete slider. */
+  @Input()
+  get valueIndicatorText(): string { return this._valueIndicatorText; }
+  set valueIndicatorText(v: string) {
+    this._valueIndicatorText = v;
+    this._cdr.detectChanges();
+  }
+  private _valueIndicatorText: string;
+
+  /** Event emitted every time the cursor moves onto the MatSliderThumbDecorator. */
+  @Output() mouseenter: EventEmitter<void> = new EventEmitter<void>();
+
+  /** Event emitted every time the cursor moves away from the MatSliderThumbDecorator. */
+  @Output() mouseleave: EventEmitter<void> = new EventEmitter<void>();
+
+  /** The visible circle for the slider thumb. This reference is used by the SliderAdapter. */
+  @ViewChild('knob') _knob: ElementRef<HTMLElement>;
+
+  constructor(
+    private _cdr: ChangeDetectorRef,
+    private _elementRef: ElementRef<HTMLElement>,
+    ) {}
+
+  /** Returns the thumb that this decorator corresponds to. */
+  _getThumb(): Thumb {
+    return this._getHostElement().tagName === 'MAT-SLIDER-END-THUMB-DECORATOR'
```

Comment:
This will hit the DOM unnecessarily on every invocation. The `tagName` of an element can't change so we could resolve it once in the constructor instead.

Reply:
This method will be used in the future by the child component `MatSliderKnob` to figure out what thumb it corresponds  to. It will only need to call this method once in its constructor so I don't see it causing any performance issues.

I initially had this value calculated lazily in a getter but I was concerned that I was adding unnecessary complexity to the code for a negligible performance gain.

```ts
get thumb(): Thumb {
    if (!this._thumb) {
        this._thumb = this._getHostElement().tagName === 'MAT-SLIDER-END-THUMB-DECORATOR'
            ? Thumb.END
            : Thumb.START;
    }
    return this._thumb;
}
private _thumb: Thumb;
```

Reply:
you should call `.toUpperCase()` on `tagName`, I don't think you're guaranteed it will be all caps

Reply:
I'm waiting for your response to whether or not we're going to collapse the `MatSliderThumbDecorator` back into `MatSlider` before I add this change.